frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ManageServingRuntimeModal.tsx

```diff
@@ -96,13 +96,28 @@ const ManageServingRuntimeModal: React.FC<ManageServingRuntimeModalProps> = ({
         servingRuntime.metadata.name,
         servingRuntime.metadata.namespace,
         secrets,
+        { dryRun: true },
       ),
-      updateServingRuntime(fillData, servingRuntime, customServingRuntimesEnabled),
+      updateServingRuntime(fillData, servingRuntime, customServingRuntimesEnabled, {
+        dryRun: true,
+      }),
     ])
-      .then(() => {
-        setActionInProgress(false);
-        onBeforeClose(true);
-      })
+      .then(() =>
+        Promise.all([
+          updateSecrets(
+            fillData,
+            servingRuntime.metadata.name,
+            servingRuntime.metadata.namespace,
+            secrets,
+          ),
+          updateServingRuntime(fillData, servingRuntime, customServingRuntimesEnabled),
+        ])
+          .then(() => {
+            setActionInProgress(false);
+            onBeforeClose(true);
+          })
+          .catch((e) => setErrorModal(e)),
```

Comment:
This catch is not needed -- Promises resolve `.then` returned promises. Your .catch below here will be able to catch your values too.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#chaining

Reply:
That's great to know thanks, I wasn't aware, updating it